src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreTaskBase.cs

```diff
@@ -173,35 +226,53 @@ public override bool Execute()
         /// <summary>
         /// Gets the command-line arguments to use when launching the process that executes the restore.
         /// </summary>
-        internal IEnumerable<string> GetCommandLineArguments()
+        /// <param name="globalProperties">Receives a <see cref="Dictionary{TKey, TValue}" /> containing the global properties.</param>
+        internal string GetCommandLineArguments(Dictionary<string, string> globalProperties)
         {
+            // First get the command-line arguments including the global properties
+            return CreateArgumentString(EnumerateCommandLineArguments(SerializeGlobalProperties ? null : globalProperties));
+
+            IEnumerable<string> EnumerateCommandLineArguments(Dictionary<string, string> globalProperties)
+            {
 #if IS_CORECLR
-            // The full path to the executable for dotnet core
-            yield return Path.Combine(ThisAssemblyLazy.Value.DirectoryName, Path.ChangeExtension(ThisAssemblyLazy.Value.Name, ".Console.dll"));
+                // The full path to the executable for dotnet core
+                yield return Path.Combine(ThisAssemblyLazy.Value.DirectoryName, Path.ChangeExtension(ThisAssemblyLazy.Value.Name, ".Console.dll"));
 #endif
+                var options = GetOptions();
 
-            var options = GetOptions();
-
-            // Semicolon delimited list of options
-            yield return string.Join(";", options.Select(i => $"{i.Key}={i.Value}"));
+                // Semicolon delimited list of options
+                yield return string.Join(";", options.Select(i => $"{i.Key}={i.Value}"));
 
-            // Full path to MSBuild.exe or MSBuild.dll
+                // Full path to MSBuild.exe or MSBuild.dll
 #if IS_CORECLR
-            yield return Path.Combine(MSBuildBinPath, "MSBuild.dll");
+                yield return Path.Combine(MSBuildBinPath, "MSBuild.dll");
```

Comment:
Can we use `Path.Join` in this code path? https://learn.microsoft.com/dotnet/api/system.io.path.join?view=net-7.0

Reply:
I think I'll leave it since this is only called once and wouldn't make much of a difference.